server/src/internalClusterTest/java/org/elasticsearch/monitor/metrics/NodeIndexingMetricsIT.java

```diff
@@ -0,0 +1,275 @@
+                "es.indexing.coordinating_operations.rejections.ratio"
+            );
+            assertThat(coordinatingOperationsRejectionsRatio.getDouble(), equalTo(0.0));
+
+            var primaryOperationsSize = getRecordedMetric(plugin::getLongAsyncCounterMeasurement, "es.indexing.primary_operations.size");
+            assertThat(primaryOperationsSize.getLong(), greaterThan(0L));
+
+            var primaryOperationsTotal = getRecordedMetric(plugin::getLongAsyncCounterMeasurement, "es.indexing.primary_operations.total");
+            // Note: `delete` request goes thru `TransportBulkAction` invoking coordinating/primary limit checks
+            assertThat(primaryOperationsTotal.getLong(), equalTo((long) docsCount + deletesCount));
+
+            var primaryOperationsCurrentSize = getRecordedMetric(
+                plugin::getLongGaugeMeasurement,
+                "es.indexing.primary_operations.current.size"
+            );
+            assertThat(primaryOperationsCurrentSize.getLong(), equalTo(0L));
+
+            var primaryOperationsCurrentTotal = getRecordedMetric(
+                plugin::getLongGaugeMeasurement,
+                "es.indexing.primary_operations.current.total"
+            );
+            assertThat(primaryOperationsCurrentTotal.getLong(), equalTo(0L));
+
+            var primaryOperationsRejectionsTotal = getRecordedMetric(
+                plugin::getLongAsyncCounterMeasurement,
+                "es.indexing.primary_operations.rejections.total"
+            );
+            assertThat(primaryOperationsRejectionsTotal.getLong(), equalTo(0L));
+
+            var primaryOperationsRejectionsRatio = getRecordedMetric(
+                plugin::getDoubleGaugeMeasurement,
+                "es.indexing.primary_operations.rejections.ratio"
+            );
+            assertThat(primaryOperationsRejectionsRatio.getDouble(), equalTo(0.0));
+
+        });
+
+    }
+
+    public void testCoordinatingRejectionMetricsArePublishing() throws Exception {
+
+        // lower Indexing Pressure limits to trigger coordinating rejections
+        final String dataNode = internalCluster().startNode(Settings.builder().put(MAX_INDEXING_BYTES.getKey(), "1KB"));
+        ensureStableCluster(1);
+
+        final TestTelemetryPlugin plugin = internalCluster().getInstance(PluginsService.class, dataNode)
+            .filterPlugins(TestTelemetryPlugin.class)
+            .findFirst()
+            .orElseThrow();
+        plugin.resetMeter();
+
+        assertAcked(prepareCreate("test").get());
+
+        final BulkRequest bulkRequest = new BulkRequest();
+        final int batchCount = randomIntBetween(100, 1000);
+        for (int i = 0; i < batchCount; i++) {
+            bulkRequest.add(new IndexRequest("test").source("field", randomAlphaOfLength(100)));
+        }
+
+        // big batch should not pass thru coordinating limit check
+        client(dataNode).bulk(bulkRequest, new ActionListener<>() {
```

Comment:
can we use a future here and wait until the bulk responds as we do in the other test?